docs/reference/scripting/access-fields.asciidoc

```diff
@@ -0,0 +1,86 @@
+[[script-fields-api]]
+== Access fields in a document with the `field` API
+++++
+<titleabbrev>Access fields in a document</titleabbrev>
+++++
+
+beta::["The `field` API is still in development and should be considered a beta feature. The API is subject to change and this iteration is likely not the final state.",{es-issue}78920]
+
+Use the `field` API to access document fields:
+
+[source,painless]
+----
+field('my_field').get()
+----
+
+This API fundamentally changes how you access documents in Painless. Previously,
+you had to access the `doc` map with the field name that you wanted to access:
+
+[source,painless]
+----
+doc['my_field'].value
+----
+
+Accessing document fields this way didn't handle missing values or missing
+mappings, which meant that your Painless scripts had to check that fields exist
+and create intermediary objects for handling such omissions.
+
+Instead, use the `field` API, which is the preferred approach to access
+documents in Painless. The `field` API abstracts access options (such as
+`_source` or `doc_values`) and provides a field-centric view of your documents.
+
+NOTE: Some fields aren't yet compatible with the `fields` API, such as `text` or
+`geo` fields. since we are still trying to figure out the best values there to return, so continued use of doc there is what is available.
```

Comment:
~since we are still trying to figure out the best values there to return~
So continue to use doc for the few fields that are not supported in the new API.

Reply:
I removed that last bit in a previous commit, and would like to rewrite this as:

>NOTE: Some fields aren't yet compatible with the `fields` API, such as `text` or `geo` fields. Continue using `doc` to access field types that the `field` API doesn't support.